### PR TITLE
Update to retrieve results with v0.3

### PR DESCRIPTION
**Pull request by lillian542**
This makes retrieving simulator results with the v0.3 interface work. 

I have no hardware access and we run no hardware tests, so I can't promise that this works for hardware.

### CODE REVIEW

File: pennylane_ionq/api_client.py [hunk 1]
```diff
@@ -449,7 +455,6 @@ def __init__(self, client=None, api_key=None):
         """
         self.fields = (
             Field("id", str),
-            Field("type", str),
```
Comment (timmysilv) [inline pennylane_ionq/api_client.py:452]: why is this removed?

File: pennylane_ionq/device.py [hunk 2]
```diff
@@ -231,7 +231,7 @@ def _submit_job(self):
         # state (as a base-10 integer string) to the probability
         # as a floating point value between 0 and 1.
         # e.g., {"0": 0.413, "9": 0.111, "17": 0.476}
-        self.histogram = job.data.value["histogram"]
+        self.histogram = job.data.value
```
Comment (timmysilv) [inline pennylane_ionq/device.py:234]: is the job data value now the histogram?

**Review by timmysilv — COMMENTED** [review]
just a couple of questions. also do the device integration tests run on hardware? anyway, they passed so at least there's that

File: pennylane_ionq/device.py (hunk 2 above)
Comment (lillian542) [reply pennylane_ionq/device.py:234]: Yes, it used to be that one of the returned fields in the result of querying a task was `{..., data: {'histogram': {states and probs here}, 'registers': None}, ...}`. Now data is retrieved from the `results` url, and it only includes `{states and probs here}`.

**Comment by lillian542** [comment]
> also do the device integration tests run on hardware?
The answer to this question is always no

File: pennylane_ionq/api_client.py [hunk 3]
```diff
@@ -350,6 +352,10 @@ def refresh_data(self, data):
         for field in self.resource.fields:
             field.set(data.get(field.name, None))
 
+        if "results_url" in data.keys():
+            result = self.client.get(self.join_path(data["results_url"]), params=params)
+            self.resource.fields[-1].set(result.json())
```
Comment (timmysilv) [inline pennylane_ionq/api_client.py:357]: I suppose `self.resource` is a `Job` type, and `.fields[-1]` is the `data` field? if so, can we ensure this more robustly, or at least leave a comment in the `Job` field definitions ensuring that `data` remains last?

Comment (lillian542) [reply pennylane_ionq/api_client.py:357]: Good point - I'm not sure how to do it more robustly without changing the format of what is returned for `fields` (which would maybe be fine, but I'm trying to change as little as possible), but I added a comment in the fields definition

**Review by timmysilv — APPROVED** [review]
thanks for figuring this all out 🏆
